processing/src/main/java/org/apache/carbondata/processing/loading/steps/CarbonRowDataWriterProcessorStepImpl.java

```diff
@@ -212,7 +212,11 @@ private void finish(CarbonFactHandler dataHandler, int iteratorIndex) {
     try {
       processingComplete(dataHandler);
     } catch (CarbonDataLoadingException e) {
-      exception = new CarbonDataWriterException(e.getMessage(), e);
+      // only assign when exception is null
+      // else it will erase original root cause
+      if (null == exception) {
```

Comment:
Why should we keep this exception？If we only want to do some statistics, we can add that code in finally code block and you can just throw the exception in catch code block